tests/test_image.py

```diff
@@ -129,3 +138,76 @@ def testImageFlip(self):
         for axis in axes:
             with self.assertRaisesRegex(ValueError, "stack axis"):
                 _ = self.im.flip(axis)
+
+    def testShape(self):
+        self.assertEqual(self.ims.shape, self.ims_np.shape)
+        self.assertEqual(self.ims.stack_shape, self.ims_np.shape[:-2])
+        self.assertEqual(self.ims.stack_ndim, 1)
+
+    def testMultiDimShape(self):
+        self.assertEqual(self.mdim_ims.shape, self.mdim_ims_np.shape)
+        self.assertEqual(self.mdim_ims.stack_shape, self.mdim_ims_np.shape[:-2])
+        self.assertEqual(self.mdim_ims.stack_ndim, self.mdim)
+        self.assertEqual(self.mdim_ims.n_images, self.mdim * self.ims.n_images)
+
+    def testMultiDimGets(self):
+        for X in self.mdim_ims:
+            self.assertTrue(np.allclose(self.ims_np, X))
+
+        # Test a slice
+        self.assertTrue(np.allclose(self.mdim_ims[:, 1:], self.ims[1:]))
+
+    def testMultiDimSets(self):
+        self.mdim_ims[0, 1] = 123
+        # Check the values changed
+        self.assertTrue(np.allclose(self.mdim_ims[0, 1], 123))
+        # and only those values changed
+        self.assertTrue(np.allclose(self.mdim_ims[0, 0], self.ims_np[0]))
+        self.assertTrue(np.allclose(self.mdim_ims[0, 2:], self.ims_np[2:]))
+        self.assertTrue(np.allclose(self.mdim_ims[1, :], self.ims_np))
+
+    def testMultiDimSetsSlice(self):
+        # Test setting a slice
+        self.mdim_ims[0, 1:] = 456
+        # Check the values changed
+        self.assertTrue(np.allclose(self.mdim_ims[0, 1:], 456))
+        # and only those values changed
+        self.assertTrue(np.allclose(self.mdim_ims[0, 0], self.ims_np[0]))
+        self.assertTrue(np.allclose(self.mdim_ims[1, :], self.ims_np))
+
+    def testMultiDimReshape(self):
+        # Try mdim reshape
+        X = self.mdim_ims.stack_reshape(*self.mdim_ims.stack_shape[::-1])
+        self.assertEqual(X.stack_shape, self.mdim_ims.stack_shape[::-1])
+
+    def testMultiDimFlattens(self):
+        # Try flattening
+        X = self.mdim_ims.stack_reshape(self.mdim_ims.n_images)
+        self.assertEqual(X.stack_shape, (self.mdim_ims.n_images,))
+
+    def testMultiDimFlattensTrick(self):
+        # Try flattening with -1
+        X = self.mdim_ims.stack_reshape(-1)
+        self.assertEqual(X.stack_shape, (self.mdim_ims.n_images,))
+
+    def testMultiDimReshapeTuples(self):
+        # Try flattening with (-1,)
+        X = self.mdim_ims.stack_reshape((-1,))
+        self.assertEqual(X.stack_shape, (self.mdim_ims.n_images,))
+
+        # Try mdim reshape
+        X = self.mdim_ims.stack_reshape(self.mdim_ims.stack_shape[::-1])
+        self.assertEqual(X.stack_shape, self.mdim_ims.stack_shape[::-1])
+
```

Comment:
Since stack reshaping is such a core feature, maybe we should apply the same reshaping operation to the `mdim_ims_np` and compare it with the resulting `Image`'s `asnumpy()` after reshaping?